pkg/workflows/wasm/host/module.go

```diff
@@ -236,6 +268,17 @@ func NewModule(modCfg *ModuleConfig, binary []byte) (*Module, error) {
 		return nil, err
 	}
 
+	if modCfg.Determinism != nil {
```

Comment:
@MStreet3 Can we put all wasi overrides in newWasiLinker? Your code as written has some coupling in it, in that it relies on AllowShadowing to be set in the newWasiLinker function